Lib/timeit.py

```diff
@@ -243,6 +244,66 @@ def repeat(stmt="pass", setup="pass", timer=default_timer,
     return Timer(stmt, setup, timer, globals).repeat(repeat, number)
 
 
+def run(stmt="pass", setup="pass", timer=default_timer,
+        repeat=default_repeat, number=None,
+        time_unit=None, verbose=False):
+    precision = 3
+    if verbose:
+        precision += 1
+
+    t = Timer(stmt, setup, timer)
+    if number == None:
+        # determine number so that 0.2 <= total time < 2.0
+        callback = None
+        if verbose:
+            def callback(number, time_taken):
+                msg = "{num} loop{s} -> {secs:.{prec}g} secs"
+                plural = (number != 1)
+                print(msg.format(num=number, s='s' if plural else '',
+                                 secs=time_taken, prec=precision))
+
+        number, _ = t.autorange(callback)
+
+        if verbose:
+            print()
+
+    raw_timings = t.repeat(repeat, number)
+
+    def format_time(dt):
+        unit = time_unit
+
+        if unit is not None:
+            scale = units[unit]
+        else:
+            scales = [(scale, unit) for unit, scale in units.items()]
+            scales.sort(reverse=True)
+            for scale, unit in scales:
+                if dt >= scale:
+                    break
+
+        return "%.*g %s" % (precision, dt / scale, unit)
+
+    if verbose:
+        print("raw times: %s" % ", ".join(map(format_time, raw_timings)))
+        print()
+    timings = [dt / number for dt in raw_timings]
+
+    best = min(timings)
+    print("%d loop%s, best of %d: %s per loop"
```

Comment:
We're calculating best twice:

```suggestion
    best = min(timings)
    worst = max(timings)
    print("%d loop%s, best of %d: %s per loop"
```